src/main/java/sparksoniq/jsoniq/runtime/iterator/SparkRuntimeIterator.java

```diff
@@ -58,10 +58,7 @@ public void close(){
     }
 
     @Override
-    public Item next(){
-        if(!this._isOpen)
-            throw new IteratorFlowException("Runtime iterator is not open", getMetadata());
-
+    public boolean hasNext(){
```

Comment:
Note that this code was copied over to a new PR, and tests were added there to make sure there is no regression on this later on.